### PR TITLE
Fix save short cut

### PR DESCRIPTION
**Pull request by duanxianbo**
[(task:21427)](https://staging.teamyapp.com/teams/107/tasks/21427)

### CODE REVIEW

File: src/components/internal/Modals/TaskDetailModal.component.tsx [hunk 1]
```diff
@@ -312,7 +312,7 @@ export class TaskDetailModalComponent extends Component<Props, State> {
                     return;
                 }
 
-                const { key, eventType, event } = keyboardEvent;
+                const { key, eventType } = keyboardEvent;
```
Comment (naturetime10) [inline src/components/internal/Modals/TaskDetailModal.component.tsx]: Can we remove this loop completely?

**Review by naturetime10 — CHANGES_REQUESTED** [review]

**Review by naturetime10 — CHANGES_REQUESTED** [review]
Left a comment

File: src/components/internal/Modals/TaskDetailModal.component.tsx (hunk 1 above)
Comment (duanxianbo) [reply src/components/internal/Modals/TaskDetailModal.component.tsx]: We still need to check if the cmd key is pressed

File: src/components/internal/Modals/TaskDetailModal.component.tsx [hunk 2]
```diff
@@ -390,6 +360,35 @@ export class TaskDetailModalComponent extends Component<Props, State> {
...
     };
 
+    private onKeyDown = (event: KeyboardEvent) => {
+        const platform = getPlatform();
+        if (
+            (platform === 'Mac' && event.metaKey) ||
+            (platform === 'Windows' && event.key === 'Control')
+        ) {
+            this.setState({
+                isCmdKeyPressed: true,
```
Comment (naturetime10) [inline src/components/internal/Modals/TaskDetailModal.component.tsx:369]:
```suggestion
                isCmdKeyPressed: true,
                isControlKeyPressed: true
```

File: src/components/internal/Modals/TaskDetailModal.component.tsx [hunk 3]
```diff
@@ -390,6 +360,35 @@ export class TaskDetailModalComponent extends Component<Props, State> {
...
+            });
+        }
+
+        if (event.key === 's' && this.state.isCmdKeyPressed) {
```
Comment (naturetime10) [inline src/components/internal/Modals/TaskDetailModal.component.tsx]:
```suggestion
        if (event.key === 's' && (this.state.isCmdKeyPressed || this.state. isControlKeyPressed)) {
```

File: src/components/internal/Modals/TaskDetailModal.component.tsx (hunk 2 above)
Comment (naturetime10) [reply src/components/internal/Modals/TaskDetailModal.component.tsx:369]: Let's separately check control and command key and save them into state.

File: src/components/internal/Modals/TaskDetailModal.component.tsx [hunk 4]
```diff
@@ -390,6 +360,35 @@ export class TaskDetailModalComponent extends Component<Props, State> {
...
+    };
+
+    private onKeyUp = (event: KeyboardEvent) => {
+        const platform = getPlatform();
+        if (
+            (platform === 'Mac' && !event.metaKey) ||
+            (platform === 'Windows' && event.key === 'Control')
+        ) {
+            this.setState({
+                isCmdKeyPressed: false,
```
Comment (naturetime10) [inline src/components/internal/Modals/TaskDetailModal.component.tsx:391]: Let's separately check control and command key and remove them from state.

**Review by naturetime10 — CHANGES_REQUESTED** [review]

File: src/components/internal/Modals/TaskDetailModal.component.tsx (hunk 2 above)
Comment (naturetime10) [reply src/components/internal/Modals/TaskDetailModal.component.tsx:369]: Can we have separate if check for command and control keys?

File: src/components/internal/Modals/TaskDetailModal.component.tsx [hunk 5]
```diff
@@ -390,6 +362,44 @@ export class TaskDetailModalComponent extends Component<Props, State> {
...
+        if (platform === 'Windows' && event.key === 'Control') {
+            this.setState({
+                isControlKeyPressed: true,
+            });
+        }
+
+        if (
+            event.key === 's' &&
+            (this.state.isCmdKeyPressed || this.state.isControlKeyPressed)
+        ) {
```
Comment (naturetime10) [inline src/components/internal/Modals/TaskDetailModal.component.tsx]:
```suggestion
        if ((this.state.isCmdKeyPressed || this.state.isControlKeyPressed) &&
            event.key === 's'
        ) {
```

**Review by naturetime10 — APPROVED** [review]

**Pull request merged**